src/debugger/channel/MessageMarshaller.js

```diff
@@ -294,24 +306,24 @@ export class MessageMarshaller {
     }
   }
 
-  _unmarshallBreakpointStoppedResponse(requestID: number, parts: Array<string>): DebuggerResponse {
-    invariant(parts.length === 3, "Incorrect number of arguments in breakpoint stopped response");
-    let filePath = parts[0];
-    let line = parseInt(parts[1], 10);
-    invariant(!isNaN(line), "Invalid line number");
-    let column = parseInt(parts[2], 10);
-    invariant(!isNaN(column), "Invalid column number");
-    let result: BreakpointStoppedResult = {
-      kind: "breakpoint-stopped",
-      filePath: filePath,
-      line: line,
-      column: column,
-    };
-    let dbgResponse: DebuggerResponse = {
-      id: requestID,
-      result: result,
-    };
-    return dbgResponse;
+  _unmarshallPrepackStoppedResponse(requestID: number, responseStr: string): DebuggerResponse {
+    try {
+      let result = JSON.parse(responseStr);
+      invariant(result.kind === "stopped");
+      invariant(result.hasOwnProperty("reason"));
+      invariant(result.hasOwnProperty("filePath"));
+      invariant(result.hasOwnProperty("line"));
+      invariant(!isNaN(result.line));
+      invariant(result.hasOwnProperty("column"));
+      invariant(!isNaN(result.column));
+      let dbgResponse: DebuggerResponse = {
+        id: requestID,
+        result: result,
+      };
+      return dbgResponse;
+    } catch (e) {
```

Comment:
This is ugly. I suggest you take out the message parsing "JSON.parse()" into a method, which needs try catch. Then all _unmarshalXXX methods do not need try...catch.

Reply:
I will do this and the refactoring of marshallXXX in separate PRs to not make this one too big.